Core/src/org/sleuthkit/autopsy/geolocation/MBTilesTileFactory.java

```diff
@@ -0,0 +1,332 @@
+            connector.closeConnection();
+            connector = null;
+        }
+    }
+
+    @Override
+    protected synchronized void startLoading(Tile tile) {
+        if (tile.isLoading()) {
+            return;
+        }
+        pendingTiles++;
+        tile.setLoading(true);
+        try {
+            tileQueue.put(tile);
+            getService().submit(new TileRunner());
+        } catch (InterruptedException ex) {
+
+        }
+    }
+
+    /**
+     * Increase the priority of this tile so it will be loaded sooner.
+     *
+     * @param tile the tile
+     */
+    synchronized void promote(Tile tile) {
+        if (tileQueue.contains(tile)) {
+            try {
+                tileQueue.remove(tile);
+                tile.setPriority(Tile.Priority.High);
+                tileQueue.put(tile);
+            } catch (InterruptedException ex) {
+
+            }
+        }
+    }
+
+    /**
+     * @return the number of pending (loading or queues) tiles
+     */
+    int getPendingTiles() {
+        return pendingTiles;
+    }
+
+    /**
+     * An inner class which actually loads the tiles. Used by the thread queue.
+     * Subclasses can override this via {@link #createTileRunner(Tile)} if
+     * necessary.
+     */
+    private class TileRunner implements Runnable {
+
+        /**
+         * Gets the full URI of a tile.
+         *
+         * @param tile the tile
+         *
+         * @throws URISyntaxException if the URI is invalid
+         * @return a URI for the tile
+         */
+        protected URI getURI(Tile tile) throws URISyntaxException {
+            if (tile.getURL() == null) {
+                return null;
+            }
+            return new URI(tile.getURL());
+        }
+
+        @Override
+        public void run() {
+            /*
+             * Attempt to load the tile from . If loading fails, retry two more
+             * times. If all attempts fail, nothing else is done. This way, if
+             * there is some kind of failure, the pooled thread can try to load
+             * other tiles.
+             */
+            final Tile tile = tileQueue.remove();
+
+            int remainingAttempts = 3;
+            while (!tile.isLoaded() && remainingAttempts > 0) {
+                remainingAttempts--;
+                try {
+                    URI uri = getURI(tile);
+                    BufferedImage img = cache.get(uri);
+                    if (img == null) {
+                        byte[] bimg = connector.getTileBytes(uri.toString());
+                        if (bimg != null && bimg.length > 0) {
+                            img = ImageIO.read(new ByteArrayInputStream(bimg));
+                            cache.put(uri, bimg, img);
+                            img = cache.get(uri);
+                        }
+                    }
+                    if (img == null) {
+                        logger.log(Level.INFO, String.format("Failed to load: %s", uri));
+                    } else {
```

Comment:
Maybe add a comment here about what might cause this so when they appear someone can have an idea why it happened.  IE: drilled down to far into map.